Microsoft.Research/ManagedContract.Setup/CSharpCodeSnippets/contractensuresnotisnullorwhitespace.snippet

```diff
@@ -0,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<CodeSnippets xmlns="http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet">
+  <CodeSnippet Format="1.0.0">
+    <Header>
+      <SnippetTypes>
+        <SnippetType>Expansion</SnippetType>
+      </SnippetTypes>
+      <Title>Contract.Ensures(!string.IsNullOrWhiteSpace(Contract.Result()))</Title>
+      <Shortcut>cesnw</Shortcut>
+      <Description>Emits an 'ensures' clause specifying result not null or white space</Description>
+      <Author>Jonathan de Halleux</Author>
+    </Header>
+    <Snippet>
+      <Imports>
+        <Import>
+          <Namespace>System.Diagnostics.Contracts</Namespace>
```

Comment:
:bulb: Rather than use this strategy, you can use a [non-editable literal](https://gist.github.com/sharwell/cb8e5bfd2c1435936c27#file-throwifargumentnull-snippet-L24-L29) to automatically reference `Contract` using the shortest available form.